tests/test_stash.py

```diff
@@ -221,3 +222,12 @@ def test_pop_stash_with_index_changes(self) -> None:
         # Verify index has the staged changes
         index = self.repo.open_index()
         self.assertIn(b"new.txt", index)
+
+
+class StashInWorktreeTest(StashTests):
```

Comment:
I can split the above `StashTests` to `BaseStashTest` and inherit from it if necessary. Let me know what you prefer.